lib/benchee/benchmark/measure/memory.ex

```diff
@@ -44,17 +55,22 @@ defmodule Benchee.Benchmark.Measure.Memory do
   defp return_memory({memory_usage, result}) when memory_usage < 0, do: {nil, result}
   defp return_memory({memory_usage, result}), do: {memory_usage, result}
 
-  defp measure_memory(fun, tracer) do
-    word_size = :erlang.system_info(:wordsize)
-    {:garbage_collection_info, heap_before} = Process.info(self(), :garbage_collection_info)
-    result = fun.()
-    {:garbage_collection_info, heap_after} = Process.info(self(), :garbage_collection_info)
-    mem_collected = get_collected_memory(tracer)
+  defp measure_memory(fun, tracer, parent) do
+    :erlang.garbage_collect()
+    send(tracer, :begin_collection)
 
-    memory_used =
-      (total_memory(heap_after) - total_memory(heap_before) + mem_collected) * word_size
+    receive do
+      :ready -> nil
+    end
 
-    {memory_used, result}
+    result = fun.()
+    send(parent, {:result, result})
+    :erlang.garbage_collect()
+    send(tracer, :end_collection)
+
+    receive do
+      :ready -> nil
```

Comment:
not sure what best practice is - but we reuse the `:ready` message here, seems cleaner/less prone to some weird race conditions or something if the message were different. Wdyt?

Reply:
I'm cool with `:ready_to_begin` and `:ready_to_finish`. It _shouldn't_ be possible for there to be any race conditions because at this point it's essentially synchronous (which is why we're blocking with `receive` while waiting for a response from the tracer), but it's always better to be safe than sorry!